fe/fe-core/src/main/java/com/starrocks/load/DeleteHandler.java

```diff
@@ -284,7 +290,30 @@ public void process(DeleteStmt stmt) throws DdlException, QueryStateException {
             LOG.info("waiting delete Job finish, signature: {}, timeout: {}", transactionId, timeoutMs);
             boolean ok = false;
             try {
-                ok = countDownLatch.await(timeoutMs, TimeUnit.MILLISECONDS);
+                long countDownTime = timeoutMs;
+                while (countDownTime > 0) {
+                    if (countDownTime > 1000) {
+                        countDownTime -= 1000;
+                        if (killJobMap.containsKey(deleteJob.getId())) {
+                            Iterator<Long> iterator = killJobMap.keySet().iterator();
+                            while (iterator.hasNext()) {
+                                if (iterator.next().equals(deleteJob.getId())) {
+                                    iterator.remove();
+                                    break;
+                                }
+                            }
+                            cancelJob(deleteJob, CancelType.USER, "user cancelled");
+                            throw new DdlException("Cancelled");
+                        }
+                        ok = countDownLatch.await(1, TimeUnit.SECONDS);
```

Comment:
You can define a constant value of 1000 and use it here, so the time unit can be same with others.